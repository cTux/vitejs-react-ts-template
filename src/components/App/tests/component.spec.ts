import { AppDriver as EnzymeAppDriver } from './driver.enzyme';
import { AppDriver as TestingLibraryAppDriver } from './driver.tl';

describe('App', () => {
  [EnzymeAppDriver, TestingLibraryAppDriver].forEach((Driver) => {
    describe(`${Driver.displayName}`, () => {
      let driver;

      beforeEach(() => {
        driver = new Driver();
      });

      it('should render w/o errors', () => {
        expect(() => driver.render()).not.toThrow();
      });

      it('should have default view', async () => {
        await driver.render();
        expect(driver.component.get.html()).toMatchSnapshot();
      });
    });
  });
});
