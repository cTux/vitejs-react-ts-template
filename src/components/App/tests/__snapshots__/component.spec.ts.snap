// Vitest Snapshot v1

exports[`App > EnzymeAppDriver > should have default view 1`] = `"<div class=\\"HG3jz\\">Loading...</div>"`;

exports[`App > TestingLibraryAppDriver > should have default view 1`] = `"<div><div class=\\"Xzv-g\\"><h1>app.title</h1><button>btn.counter 0</button><div class=\\"s1pQH\\"><button data-qa=\\"btn-en\\">btn.switch-to.en</button><button data-qa=\\"btn-de\\">btn.switch-to.de</button></div></div></div><div><div class=\\"Xzv-g\\"><h1>app.title</h1><button>btn.counter 0</button><div class=\\"s1pQH\\"><button data-qa=\\"btn-en\\">btn.switch-to.en</button><button data-qa=\\"btn-de\\">btn.switch-to.de</button></div></div></div>"`;
